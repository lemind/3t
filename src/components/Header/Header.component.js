import React from 'react';
import './Header.less';

export class HeaderComponent extends React.Component {
  constructor(props) {
    super();
    console.log('object', props);
  }

  render() {
    return (
      <header>
        <div className="menu">
          <a
            href="javascript:;"
            onClick={ () => this.props.resetGame() }
          >New game</a>
        </div>
      </header>
    )
  };
}
