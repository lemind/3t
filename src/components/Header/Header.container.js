import { connect } from 'react-redux';

import { HeaderComponent } from './Header.component';
import { processActions } from '../../redux/process/index';

export const HeaderContainer = connect(
  function mapStateToProps(state) {
    return {}
  },
  function mapDispatchToProps(dispatch) {
    return {
      resetGame: () => {
        dispatch(processActions.resetGame())
      }
    };
  }
)(HeaderComponent);