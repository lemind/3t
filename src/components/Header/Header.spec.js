import React from 'react';
import { shallow } from 'enzyme';
import { HeaderComponent } from './Header.component';

describe('Header', () => {

  beforeEach(() => {
  })

  test('should render HeaderComponent', () => {
    expect(shallow(<HeaderComponent />)).toMatchSnapshot();
  })
})
