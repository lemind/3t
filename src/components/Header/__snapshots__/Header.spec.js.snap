// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render HeaderComponent 1`] = `
<header>
  <div
    className="menu"
  >
    <a
      href="javascript:;"
      onClick={[Function]}
    >
      New game
    </a>
  </div>
</header>
`;
